master/custom/factories.py

```diff
@@ -1219,3 +1219,68 @@ def setup(self, **kwargs):
     def host_triple(props):
         build_triple = props.getProperty("build_triple")
         return build_triple.split("-")[0] + "-linux-android"
+
+
+class ValgrindBuild(UnixBuild):
+    buildersuffix = ".valgrind"
+    configureFlags = [
+        "--with-pydebug",
+        "--with-valgrind",
+        "--without-pymalloc",
+    ]
+    testFlags = [
+        "test_grammar",
```

Comment:
I am starting with this and grow slowly the list of tests as we did with the undefined behaviour sanitizer